packages/block-library/src/rss/index.php

```
@@ -0,0 +1,171 @@
+		return render_rss_error_message( __( 'Use \'Latest Posts\' block for this domain', 'gutenberg' ) );
+	}
+
+	$rss = fetch_feed( $url );
+
+	if ( is_wp_error( $rss ) ) {
+		return render_rss_error_message( '<strong>' . __( 'RSS Error:', 'gutenberg' ) . '</strong> ' . $rss->get_error_message() );
+	}
+
+	if ( ! $rss->get_item_quantity() ) {
+		$rss->__destruct();
+		unset( $rss );
+		return render_rss_error_message( __( 'An error has occurred, which probably means the feed is down. Try again later.', 'gutenberg' ) );
+	}
+
+	$items = (int) $attributes['itemsToShow'];
+	if ( $items < 1 || 10 < $items ) {
+		$items = 5;
+	}
+
+	$list_items = '';
+	foreach ( $rss->get_items( 0, $items ) as $item ) {
+		$link = $item->get_link();
+
+		while ( stristr( $link, 'http' ) != $link ) {
+			$link = substr( $link, 1 );
+		}
+
+		$title = esc_html( trim( strip_tags( $item->get_title() ) ) );
+		if ( empty( $title ) ) {
+			$title = __( '(Untitled)', 'gutenberg' );
+		}
+
+		$link = esc_url( strip_tags( $link ) );
+		if ( $link ) {
+			$title = "<a href='$link'>$title</a>";
+		}
+
+		$title = "<div class='wp-block-rss__item-title'>$title</div>";
+
+		$date = '';
+		if ( $attributes['displayDate'] ) {
+			$date = $item->get_date( 'U' );
+
+			if ( $date ) {
+				$date = sprintf(
+					'<time datetime="%1$s" class="wp-block-rss__item-publish-date">%2$s</time>',
+					date_i18n( get_option( 'c' ), $date ),
+					date_i18n( get_option( 'date_format' ), $date )
+				);
+			}
+		}
+
+		$author = '';
+		if ( $attributes['displayAuthor'] ) {
+			$author = $item->get_author();
+			if ( is_object( $author ) ) {
+				$author = $author->get_name();
+				$author = '<span class="wp-block-rss__item-author">' . __( 'By', 'gutenberg' ) . ' ' . esc_html( strip_tags( $author ) ) . '</span>';
+			}
+		}
+
+		$excerpt = '';
+		if ( $attributes['displayExcerpt'] ) {
+			$excerpt = @html_entity_decode( $item->get_description(), ENT_QUOTES, get_option( 'blog_charset' ) );
```

@gziolo This line is causing the PHP waring:
````
Silencing errors is discouraged
Generic.PHP.NoSilencedErrors.Discouraged)
````

I copied this function from the widget:
https://github.com/WordPress/wordpress-develop/blob/6b366c6620fdd5960cedcdf80955966a715efa82/src/wp-includes/widgets/class-wp-widget-rss.php#L70

---

I can't think of any reason we ought to be suppressing errors here. Best to just remove the `@`, I think.